docs: remove note on disk space for caching (#5534)
aiidateam/aiida-core: docs/source/topics/provenance/caching.rst

   While AiiDA's hashes include the version of the Python package containing the calculation/data classes, it cannot detect cases where the underlying Python code was changed without increasing the version number.
   Another scenario that can lead to an erroneous cache hit is if the parser and calculation are not implemented as part of the same Python package, because the calculation nodes store only the name, but not the version of the used parser.

#. While caching saves unnecessary computations, it does not directly prevent duplication of data: the cached calculation and its output nodes are duplicated.

>    In practice, however, AiiDA's file repository implementation will detect that any files associated with these nodes are already present and simply point to those, reducing duplication to metadata stored at the database level.

Author: thanks, I wanted to phrase it slightly differently but I've incorporated your suggestions

   While AiiDA's hashes include the version of the Python package containing the calculation/data classes, it cannot detect cases where the underlying Python code was changed without increasing the version number.
   Another scenario that can lead to an erroneous cache hit is if the parser and calculation are not implemented as part of the same Python package, because the calculation nodes store only the name, but not the version of the used parser.

> #. While caching saves unnecessary computations, it does lead result in duplication of the cached calculation and its output nodes in the provenance graph.
>    At the same time, AiiDA's default disk-objectstore storage backend comes with automatic de-duplication at the object level.

Reviewer: ```suggestion
#. While caching saves unnecessary computations, it does not necessarily save space as the cached calculation and its output nodes are duplicated in the provenance graph.
   However, AiiDA's default disk-objectstore storage backend comes with automatic de-duplication at the object level.
```
